soroban-sdk/src/string.rs

```diff
@@ -65,10 +65,6 @@ impl PartialOrd for String {
 
 impl Ord for String {
     fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-        #[cfg(not(target_family = "wasm"))]
-        if !self.env.is_same_env(&other.env) {
-            return ScVal::from(self).cmp(&ScVal::from(other));
-        }
```

Comment:
Why are the comparison functions changing?

Reply:
That's unrelated to the sketch, I just didn't want to refactor the SDK to support https://github.com/stellar/rs-soroban-env/pull/1516.